From ad24a929bdcc15abae14a64ea21b821bcd8cb030 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 11 May 2011 15:19:11 +0200
Subject: [PATCH] ru_maxrss is in kilobytes on Linux

Since 2.6.32 Linux returns ru_maxrss in kilobytes. Not in pages.

See http://lists.gnu.org/archive/html/bug-gnu-utils/2008-12/msg00047.html
for discussion.
---
 time.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/time.c b/time.c
index d15fee4..43aec0b 100644
--- a/time.c
+++ b/time.c
@@ -395,7 +395,11 @@ summarize (fp, fmt, command, resp)
 		       ptok ((UL) resp->ru.ru_ixrss) / MSEC_TO_TICKS (v));
 	      break;
 	    case 'M':		/* Maximum resident set size.  */
+#ifdef __linux__
+	      fprintf (fp, "%ld", resp->ru.ru_maxrss);
+#else
 	      fprintf (fp, "%lu", ptok ((UL) resp->ru.ru_maxrss));
+#endif
 	      break;
 	    case 'O':		/* Outputs.  */
 	      fprintf (fp, "%ld", resp->ru.ru_oublock);
-- 
1.7.4.4

